providers/standard/src/airflow/providers/standard/operators/trigger_dagrun.py

```diff
@@ -209,14 +209,37 @@ def execute(self, context: Context):
                 run_after=parsed_logical_date or timezone.utcnow(),
             )
 
+        if AIRFLOW_V_3_0_PLUS:
+            self._trigger_dag_af_3(context=context, run_id=run_id, parsed_logical_date=parsed_logical_date)
+        else:
+            self._trigger_dag_af_2(context=context, run_id=run_id, parsed_logical_date=parsed_logical_date)
+
+    def _trigger_dag_af_3(self, context, run_id, parsed_logical_date):
+        from airflow.exceptions import DagRunTriggerException
+
+        raise DagRunTriggerException(
+            trigger_dag_id=self.trigger_dag_id,
+            dag_run_id=run_id,
+            conf=self.conf,
+            logical_date=parsed_logical_date,
+            reset_dag_run=self.reset_dag_run,
+            skip_when_already_exists=self.skip_when_already_exists,
+            wait_for_completion=self.wait_for_completion,
+            allowed_states=self.allowed_states,
+            failed_states=self.failed_states,
+            poke_interval=self.poke_interval,
+        )
+
+        # TODO: Support deferral
+
+    def _trigger_dag_af_2(self, context, run_id, parsed_logical_date):
         try:
             dag_run = trigger_dag(
                 dag_id=self.trigger_dag_id,
                 run_id=run_id,
                 conf=self.conf,
                 logical_date=parsed_logical_date,
                 replace_microseconds=False,
-                triggered_by=DagRunTriggeredByType.OPERATOR,
```

Comment:
Why did this get removed?

Reply:
This is Airflow 2 block, `DagRunTriggeredByType` isn't part of Airflow 2.x, only Airflow 3.

Since the tests were never executed for TriggerDagRunOperator on Airflow 2, we never caught it.